src/engine/Engine.ts

```diff
@@ -232,8 +232,8 @@ export class Engine extends Class {
     * Default [[IEngineOptions]]
     */
    private static _DefaultEngineOptions: IEngineOptions = {
-      width: 0,
-      height: 0,
+      canvasWidth: 0,
```

Comment:
Let's avoid changing the IEngineOption property. It keeps the constructor option back terse and we don't need to change every file.

I think it's more important when accessing `ex.Engine.canvasWidth` and `ex.Engine.canvasHeight` which gives us the canvas pixel width/height  vs  `ex.Engine.getWidth()` and `ex.Engine.getHeight()` which get's the current drawing surface in pixels taking zoom into account.

Perhaps it would also be good to further disambiguate the different heights in #591, and refactor `ex.Engine.getWidth()` and `ex.Engine.getHeight()` to be something like `getScreenWidth()`,`getPixelWidth()`. @jedeen What do you think?

Other than that I like the PR.